public/app/api/clients/folder/v1beta1/hooks.ts

```diff
@@ -252,3 +288,28 @@ function getUserKeys(resultFolder: ReturnType<typeof useGetFolderQuery>): string
       ].filter((v) => v !== undefined)
     : [];
 }
+
+const appPlatformFolderToLegacyFolder = (
+  folder: Folder
+): Omit<FolderDTO, 'parents' | 'canSave' | 'canEdit' | 'canAdmin' | 'canDelete' | 'createdBy' | 'updatedBy'> => {
+  // Omits properties that we can't easily get solely from the app platform response
+  // In some cases, these properties aren't used on the response of the hook,
+  // so it's best to discourage from using them anyway
+
+  const { annotations, name, creationTimestamp, generation, labels } = folder.metadata;
+  const { title } = folder.spec;
+  return {
+    id: parseInt(labels?.[DeprecatedInternalId] || '0', 10) || 0,
+    uid: name!,
```

Comment:
Could we avoid non-null assertions? Maybe by assigning the name and folder an empty string as default value when de-structuring?